packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -434,7 +434,24 @@ export const doesSectionHaveTrailingZeros = <TDate>(
 
   // We can't use `changeSectionValueFormat`, because  `utils.parse('1', 'YYYY')` returns `1971` instead of `1`.
   if (dateSectionName === 'year') {
-    return utils.formatByString(utils.setYear(utils.date()!, 1), format).length > 1;
+    const formatted2001 = utils.formatByString(utils.setYear(utils.date()!, 2001), format);
```

Comment:
Not sure if that's the best approach
The challenge here is to know if we have trailing zeroes or not on the year.
But a year format can be 2-digit or 4-digit.

Reply:
Seems like this is a "necessary evil" to cover the different year formats adapters can have. 🙈